src/bindings.rs

```diff
@@ -643,32 +644,173 @@ publish = false
     }
 }
 
-fn export_world_key_path(resolve: &Resolve, key: &WorldKey) -> String {
-    match key {
-        WorldKey::Name(name) => format!("bindings::exports::{name}", name = name.to_snake_case(),),
-        WorldKey::Interface(id) => {
-            let iface = &resolve.interfaces[*id];
-            let pkg = &resolve.packages[iface.package.expect("interface should have a package")];
-            let name = iface.name.as_deref().expect("unnamed interface");
-            format!(
-                "bindings::exports::{ns}::{pkg}::{name}",
-                ns = pkg.name.namespace.to_snake_case(),
-                pkg = pkg.name.name.to_snake_case(),
-                name = name.to_snake_case(),
+/// Represents a node in a "use" trie.
+#[derive(Default)]
+struct UseTrieNode {
+    // Map of segment index to trie node
+    children: HashMap<usize, UseTrieNode>,
+    // Set of type indexes that are used at this node
+    tys: HashSet<usize>,
+}
+
+impl UseTrieNode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>, trie: &UseTrie) -> fmt::Result {
+        // Print the children first
+        for (i, (segment, child)) in self.children.iter().enumerate() {
+            if i > 0 {
+                write!(f, ", ")?;
+            }
+
+            write!(f, "{segment}::{{", segment = trie.segments[*segment])?;
+            child.fmt(f, trie)?;
+            write!(f, "}}")?;
+        }
+
+        if !self.children.is_empty() && !self.tys.is_empty() {
+            write!(f, ", ")?;
+        }
+
+        // Next, print the types at this node
+        for (i, type_index) in self.tys.iter().enumerate() {
+            if i > 0 {
+                write!(f, ", ")?;
+            }
+
+            write!(f, "{ty}", ty = trie.types[*type_index])?;
+        }
+
+        Ok(())
+    }
+}
+
+/// A trie that maintains uses of types when generating Rust source code.
+#[derive(Default)]
+struct UseTrie {
+    /// The root trie node.
+    root: UseTrieNode,
+    /// The set of path segments.
+    segments: IndexSet<String>,
+    /// The set of type names.
+    types: IndexSet<String>,
+}
```

Comment:
Would it be feasible to move `UseTrie` and `UseTrieNode` into a separate source file? That'd keep src/bindings.rs a little tidier for people studying how bindings generation works at a high level.